nengo/conftest.py

```diff
@@ -210,12 +231,17 @@ def pytest_runtest_setup(item):
                 ('logger', 'logs', "logs not requested")]:
             if fixture_name in item.fixturenames:
                 if item.config.getvalue(option):
-                    skip = False
+                    break
                 else:
                     skipreasons.append(message)
-        if skip:
+        else:
             pytest.skip(" and ".join(skipreasons))
 
+    if 'Simulator' in item.fixturenames:
+        for test, reason in _Simulator.unsupported:
```

Comment:
Hm, compiling the regexes beforehand would complicate things a little bit and `re.match` apparently caches a certain number of recent regexes (how big is that number? I have no idea).

So I'm not sure if that's worth it.